src/Pages/ForgotPassword.cshtml.cs

```diff
@@ -0,0 +1,94 @@
+using System;
+using System.ComponentModel.DataAnnotations;
+using ePiggyWeb.DataBase;
+using ePiggyWeb.Utilities;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+
+namespace ePiggyWeb.Pages
+{
+    public class ForgotPasswordModel : PageModel
+    {
+        [Required]
+        [BindProperty]
+        public string EnteredCode { get; set; }
+
+        [Required]
+        [RegularExpression(@"^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*?&])[A-Za-z\d@$!%*?&]{6,}$", ErrorMessage = "Password must contain at least one uppercase letter, at least one number, special character and be longer than six characters.")]
+        [BindProperty]
+        [DataType(DataType.Password)]
+        public string Password { get; set; }
+
+        [Required]
+        [BindProperty]
+        [DataType(DataType.Password)]
+        public string PasswordConfirm { get; set; }
+
+        [BindProperty]
+        public string Email { get; set; }
+
+        public string ErrorMessage = "";
+
+        public bool CodeSent = false;
```

Comment:
Resharper is telling me it's unnecessary to set CodeSent default value.

Reply:
I guess bools by default are assigned to false.